src/webgpu/api/operation/memory_sync/texture/texture_sync_test.ts

```diff
@@ -101,37 +111,64 @@ export const kAllWriteOps = [
 ] as const;
 export type WriteOp = typeof kAllWriteOps[number];
 
-export const kAllReadOps = [
-  't2b-copy',
-  't2t-copy',
-  'attachment-load',
-  'storage',
-  'sample',
-] as const;
+export const kAllReadOps = ['t2b-copy', 't2t-copy', 'storage', 'sample'] as const;
 export type ReadOp = typeof kAllReadOps[number];
 
 export type Op = ReadOp | WriteOp;
 
 interface OpInfo {
   readonly contexts: OperationContext[];
+  readonly readUsage: GPUTextureUsageFlags;
+  readonly writeUsage: GPUTextureUsageFlags;
   // Add fields as needed
 }
 
 /**
  * Mapping of Op to the OperationContext(s) it is valid in
  */
-const kOpInfo: {
+export const kOpInfo: {
   readonly [k in Op]: OpInfo;
 } = /* prettier-ignore */ {
-  'write-texture': { contexts: [ 'queue' ] },
-  'b2t-copy': { contexts: [ 'command-encoder' ] },
-  't2t-copy': { contexts: [ 'command-encoder' ] },
-  't2b-copy': { contexts: [ 'command-encoder' ] },
-  'storage': { contexts: [ 'compute-pass-encoder', 'render-pass-encoder', 'render-bundle-encoder' ] },
-  'sample': { contexts: [ 'compute-pass-encoder', 'render-pass-encoder', 'render-bundle-encoder' ] },
-  'attachment-store': { contexts: [ 'render-pass-encoder' ] },
-  'attachment-resolve': { contexts: [ 'render-pass-encoder' ] },
-  'attachment-load': { contexts: [ 'render-pass-encoder' ] },
+  'write-texture': {
+    contexts: [ 'queue' ],
+    readUsage: 0,
```

Comment:
this is a bit silly, but would `get readUsage() { unreachable(); }` work?

Reply:
Nope. Complains that void is not compatible with number